src/Metadata/Property/SubresourceMetadata.php

```diff
@@ -54,4 +58,30 @@ public function withCollection(bool $collection): self
 
         return $metadata;
     }
+
+    public function withProperty(string $property): self
+    {
+        $metadata = clone $this;
+        $metadata->property = $property;
+
+        return $metadata;
+    }
+
+    public function getProperty(): string
+    {
+        return $this->property;
+    }
+
+    public function withParentResourceClass(string $parentResourceClass): self
+    {
+        $metadata = clone $this;
+        $metadata->parentResourceClass = $parentResourceClass;
+
+        return $metadata;
+    }
+
+    public function getParentResourceClass(): string
+    {
+        return $this->parentResourceClass;
+    }
```

Comment:
I'm not sure we need those informations here. Usually when using a `SubresourceMetadata` we already have those informations. I want to update the PR to try and remove those adds.